### PR TITLE
Add `validate_flow` operation to FlowsClient

### PR DESCRIPTION
**Pull request by ada-globus**
I added a quick copyedit commit to normalize capitalization of "flow" and "run", which appeared to be inconsistent. It's not per se part of the scope, but was very fast and relevant to matching the new docstring for `validate_flow`. If it's a significant barrier to getting this merged, I can break it into a separate PR.

----
📚 Documentation preview 📚: https://globus-sdk-python--979.org.readthedocs.build/en/979/

### CODE REVIEW

File: src/globus_sdk/services/flows/client.py [hunk 1]
```diff
@@ -46,14 +46,14 @@ def create_flow(
         additional_fields: dict[str, t.Any] | None = None,
     ) -> GlobusHTTPResponse:
         """
-        Create a Flow
+        Create a flow
```
Comment (derek-globus) [inline src/globus_sdk/services/flows/client.py:49]: 😬

**Review by derek-globus — APPROVED** [review]

File: src/globus_sdk/services/flows/client.py (hunk 1 above)
Comment (sirosen) [reply src/globus_sdk/services/flows/client.py:49]: We need a style-guide for this. We started using emphasis (boldface, IIRC?) in some of the docs for `flow`, `run`, `timer`, and other proper nouns which are easily confused with service names.

**Review by sirosen — COMMENTED** [review]
I'd like us to hold this change until there's a Flows service release to put that validate route out there. When the SDK or CLI "run ahead", it produces a slightly strange situation if we release.

File: src/globus_sdk/services/flows/client.py (hunk 1 above)
Comment (ada-globus) [reply src/globus_sdk/services/flows/client.py:49]: Yeah, I think the way I'd phrase it is roughly:

> Entities (e.g., flow, run, action, action provider) are lowercase and first appearance is set off as a "term" in whatever paradigm a given document uses; for the docs site, we use bold

Is there such a paradigm—or related prior-art—in the SDK docs? If so, happy to make consistent!

**Pull request merged**